fuzzing/private/oss_fuzz/repository.bzl

```diff
@@ -0,0 +1,110 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Repository rule for configuring the OSS-Fuzz engine and instrumentation."""
+
+def _to_list_repr(elements):
+    return ", ".join([repr(element) for element in elements])
+
+def _extract_build_params(
+        repository_ctx,
+        fuzzing_engine_library,
+        cflags,
+        cxxflags):
+    stub_srcs = []
+    stub_linkopts = []
+    instrum_conlyopts = []
+    instrum_cxxopts = []
+
+    if fuzzing_engine_library:
+        if fuzzing_engine_library.startswith("-"):
+            # This is actually a flag, add it to the linker flags.
+            stub_linkopts.append(fuzzing_engine_library)
+        elif fuzzing_engine_library.endswith(".a"):
+            repository_ctx.symlink(
+                repository_ctx.path(fuzzing_engine_library),
+                "oss_fuzz_engine.a",
+            )
+            stub_srcs.append("oss_fuzz_engine.a")
+        else:
+            fail("Unsupported $LIB_FUZZING_ENGINE value '%s'" % fuzzing_engine_library)
+    for cflag in cflags:
+        if cflag == "-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION":
```

Comment:
i think these and in cxxflags should be removed (this reminds me Envoy's build can remove this flag now. we originally had it because a 3p dependency built with this flag would break our fuzzers, and it was a workaround.)